src/inference.jl

```diff
@@ -1986,3 +1408,397 @@ function rxinference(;
+For example:
+
+```julia
+engine = infer(
+    ...,
+    autoupdates = my_autoupdates,
+    returnvars = (:x, :τ),
+    autostart  = false
+)
+
+- ### `predictvars`
+
+`predictvars` specifies the variables which should be predicted. In the model definition these variables are specified
+as datavars, although they should not be passed inside data argument.
+
+Similar to `returnvars`, `predictvars` accepts a `NamedTuple` or `Dict`. There are two specifications:
+- `KeepLast`: saves the last update for a variable, ignoring any intermediate results during iterations
+- `KeepEach`: saves all updates for a variable for all iterations
+
+Example: 
+
+```julia
+result = infer(
+    ...,
+    predictvars = (
+        o = KeepLast(),
+        τ = KeepEach()
+    )
+)
+```
+
+**Note**: The `predictvars` argument is exclusive for batch setting.
+
+- ### `historyvars`
+
+`historyvars` specifies the variables of interests and the amount of information to keep in history about the posterior updates when performing streamline inference. The specification is similar to the `returnvars` when applied in batch setting.
+The `historyvars` requires `keephistory` to be greater than zero.
+
+`historyvars` accepts a `NamedTuple` or `Dict` or return var specification. There are two specifications:
+- `KeepLast`: saves the last update for a variable, ignoring any intermediate results during iterations
+- `KeepEach`: saves all updates for a variable for all iterations
+
+Example: 
+
+```julia
+result = infer(
+    ...,
+    autoupdates = my_autoupdates,
+    historyvars = (
+        x = KeepLast(),
+        τ = KeepEach()
+    ),
+    keephistory = 10
+)
+```
+
+It is also possible to set either `historyvars = KeepLast()` or `historyvars = KeepEach()` that acts as an alias and sets the given option for __all__ random variables in the model.
+
+# Example: 
+
+```julia
+result = infer(
+    ...,
+    autoupdates = my_autoupdates,
+    historyvars = KeepLast(),
+    keephistory = 10
+)
+```
+
+- ### `keep_history`
+
+Specifies the buffer size for the updates history both for the `historyvars` and the `free_energy` buffers in streamline inference.
+
+- ### `iterations`
+
+Specifies the number of variational (or loopy belief propagation) iterations. By default set to `nothing`, which is equivalent of doing 1 iteration. 
+
+- ### `free_energy` 
+
+This setting specifies whenever the `infer` function should create an observable of Bethe Free Energy (BFE) values. The BFE observable returns a new computed value for each VMP iteration.
+Note, however, that it may be not possible to compute BFE values for every model. If `free_energy = true` and `keephistory > 0` the engine exposes extra fields to access the history of the Bethe free energy updates:
+
+- `engine.free_energy_history`: Returns a free energy history averaged over the VMP iterations
```

Comment:
This is exclusive to the streamlined version, the batched version does not have those

Reply:
Addressed